pallets/api/src/messaging/benchmarking.rs

```diff
@@ -0,0 +1,274 @@
+
+			let good_message = Message::IsmpResponse {
+				commitment: commitment.clone(),
+				deposit,
+				response: Default::default(),
+			};
+
+			Messages::<T>::insert(&owner, &message_id.0, &good_message);
+			IsmpRequests::<T>::insert(&commitment, (&owner, &message_id.0));
+			message_ids.try_push(message_id.0).unwrap()
+		}
+		#[extrinsic_call]
+		Pallet::<T>::remove(RawOrigin::Signed(owner.clone()), message_ids.clone());
+
+		assert_has_event::<T>(
+			crate::messaging::Event::Removed { origin: owner, messages: message_ids.to_vec() }
+				.into(),
+		)
+	}
+
+	#[benchmark]
+	fn xcm_new_query() {
+		let owner: AccountIdOf<T> = account("Alice", 0, SEED);
+		let message_id: [u8; 32] = [0; 32];
+		let responder = Location { parents: 1, interior: Junctions::Here };
+		let timeout = <BlockNumberOf<T> as One>::one() + frame_system::Pallet::<T>::block_number();
+		let callback = Callback {
+			selector: [0; 4],
+			weight: 100.into(),
+			spare_weight_creditor: owner.clone(),
+			abi: Abi::Scale,
+		};
+
+		pallet_balances::Pallet::<T>::make_free_balance_be(&owner, u32::MAX.into());
+
+		#[extrinsic_call]
+		Pallet::<T>::xcm_new_query(
+			RawOrigin::Signed(owner.clone()),
+			message_id.clone(),
+			responder.clone(),
+			timeout,
+			Some(callback.clone()),
+		);
+
+		assert_has_event::<T>(
+			crate::messaging::Event::XcmQueryCreated {
+				origin: owner,
+				id: message_id,
+				query_id: 0,
+				callback: Some(callback),
+			}
+			.into(),
+		)
+	}
+
+	/// x: Whether a successfully executing callback is provided.
+	#[benchmark]
+	fn xcm_response(x: Linear<0, 1>) {
+		let owner: AccountIdOf<T> = account("Alice", 0, SEED);
+		let message_id: [u8; 32] = [0; 32];
+		let responder = Location { parents: 1, interior: Junctions::Here };
+		let timeout = <BlockNumberOf<T> as One>::one() + frame_system::Pallet::<T>::block_number();
+		let response = Response::ExecutionResult(None);
+
+		let callback = if x == 1 {
+			// The mock will always assume successfull callback.
+			Some(Callback {
+				selector: [0; 4],
+				weight: 100.into(),
+				spare_weight_creditor: owner.clone(),
+				abi: Abi::Scale,
+			})
+		} else {
+			None
+		};
+
+		pallet_balances::Pallet::<T>::make_free_balance_be(&owner, u32::MAX.into());
+
+		Pallet::<T>::xcm_new_query(
+			RawOrigin::Signed(owner.clone()).into(),
+			message_id.clone(),
+			responder.clone(),
+			timeout,
+			callback.clone(),
+		)
+		.unwrap();
+
+		#[extrinsic_call]
+		Pallet::<T>::xcm_response(RawOrigin::Root, 0, response.clone());
+
+		assert_has_event::<T>(
+			crate::messaging::Event::XcmResponseReceived {
+				dest: owner.clone(),
+				id: message_id,
+				query_id: 0,
+				response,
+			}
+			.into(),
+		);
+		assert!(Messages::<T>::get(&owner, &message_id).is_none());
+		assert!(XcmQueries::<T>::get(0).is_none());
+	}
+
+	/// x: Is it a get. (example: 1 = get, 0 = post)
+	/// y: the response has a callback.
+	#[benchmark]
+	fn ismp_on_response(x: Linear<0, 1>, y: Linear<0, 1>) {
```

Comment:
this is not used anywhere, as far as I can tell. Shouldn't it be placed here: https://github.com/r0gue-io/pop-node/blob/74a8a782dd2a434d02beda7a7d368b3f07fcb8e2/pallets/api/src/messaging/transports/ismp.rs#L176

see here as another reference as well: 
https://github.com/RegionX-Labs/RegionX-Node/blob/main/pallets/regions/src/lib.rs#L488

Reply:
Will be part of another task